src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -2833,38 +2717,35 @@ def create_run(self, pipeline_run: PipelineRunModel) -> PipelineRunModel:
                     )
 
             # Query stack
-            if pipeline_run.stack_id is not None:
-                stack = session.exec(
-                    select(StackSchema).where(
-                        StackSchema.id == pipeline_run.stack_id
-                    )
-                ).first()
-                if stack is None:
-                    logger.warning(
-                        f"No stack with ID '{pipeline_run.stack_id}' found. "
-                        f"Creating pipeline run '{pipeline_run.name}' without "
-                        "linked stack."
-                    )
-                    pipeline_run.stack_id = None
-
-            # Query pipeline
-            pipeline = None
-            if pipeline_run.pipeline_id is not None:
-                pipeline = session.exec(
-                    select(PipelineSchema).where(
-                        PipelineSchema.id == pipeline_run.pipeline_id
-                    )
-                ).first()
-                if pipeline is None:
-                    logger.warning(
-                        f"No pipeline with ID '{pipeline_run.pipeline_id}' "
-                        f"found. Creating pipeline run '{pipeline_run.name}' "
-                        f"as unlisted run."
-                    )
-                    pipeline_run.pipeline_id = None
+            if pipeline_run.stack is None:
+                logger.warning(
+                    f"No stack found for this run. "
+                    f"Creating pipeline run '{pipeline_run.name}' without "
+                    "linked stack."
+                )
 
-            new_run = PipelineRunSchema.from_create_model(
-                run=pipeline_run, pipeline=pipeline
+            if pipeline_run.pipeline is None:
+                logger.warning(
+                    f"No pipeline found. Creating pipeline run "
+                    f"'{pipeline_run.name}' as unlisted run."
+                )
+
+            configuration = json.dumps(pipeline_run.pipeline_configuration)
+
+            new_run = PipelineRunSchema(
+                id=pipeline_run.id,
+                name=pipeline_run.name,
+                orchestrator_run_id=pipeline_run.orchestrator_run_id,
+                stack_id=pipeline_run.stack,
+                project_id=pipeline_run.project,
+                user_id=pipeline_run.user,
+                pipeline_id=pipeline_run.pipeline,
+                status=pipeline_run.status,
+                pipeline_configuration=configuration,
+                num_steps=pipeline_run.num_steps,
+                git_sha=pipeline_run.git_sha,
+                zenml_version=pipeline_run.zenml_version,
+                mlmd_id=pipeline_run.mlmd_id,
             )
```

Comment:
Similarly as above, I preferred having this in `from_create_model` or similar since it is really error-prone to do this in here in my opinion